sklearn/neighbors/regression.py

```diff
@@ -288,13 +292,28 @@ def predict(self, X):
             _y = _y.reshape((-1, 1))
 
         if weights is None:
-            y_pred = np.array([np.mean(_y[ind, :], axis=0)
-                               for ind in neigh_ind])
+
+            with warnings.catch_warnings():
+                warnings.filterwarnings('ignore', "Mean of empty slice")
+                warn_div = "invalid value encountered in true_divide"
+                warnings.filterwarnings('ignore', warn_div)
```

Comment:
(1)I don't think our code should change the default behaviour of other libraries. In this case, after user run RadiusNeighborsRegressor, he will not get any warning when he execute `np.mean([])`, it might be better to at least reset the warnings.
(2) `invalid value encountered in true_divide` seems a bit magical, could you please add some comments? Also, is there any specific reason you are blocking warnings here? It seems that you can use the same way as in the else clause. i.e.
```python
empty_obs = np.full_like(_y[0], np.nan)
y_pred = np.array([np.mean(_y[ind, :], axis=0, weights=weights[i]) if len(ind) else empty_obs
                             for (i, ind) in enumerate(neigh_ind)])
```

Reply:
The with catch_warnings context will reset when exited. But is there a reason we are not using numpy's own error manager?

Reply:
The with catch_warnings context will reset when exited. But is there a reason we are not using numpy's own error manager?

Reply:
No particular reason. I can implement the numpy warnings if you wish it to be changed.

Reply:
@jnothman Thanks a lot for the confirm. If you think blocking warnings here is the right way, I won't oppose. My main concern is the redundant tests for this edge case
@abjer If you don't have specific reason, please consider to address @jnothman's comment here. Also, please consider whether we should cut off some redundant tests.
Another thing @abjer could you please update the document to tell users what we are doing in this special case? I think it might not be so straightforward for ordinary users.

Reply:
@abjer You need to solve the conflict before this PR can be merged (e.g., merge current master into this branch). Thanks a lot for your issue and PR and also for your great patience :)